openrpc.yaml

```
@@ -252,9 +252,10 @@ methods:
       Presents a plain text signature challenge to the user and returns the
       signed response. Equivalent to `eth_sign` on some other wallets, and
       prepends a safe prefix to the signed message to prevent the challenge
-      tricking users into signing a financial transaction. This method requires
-      that the user has granted permission to interact with their account first,
-      so make sure to call `eth_requestAccounts` first.
+      tricking users into signing a financial transaction. 
+      This method requires that the user has granted permission to interact 
+      with their account first, so make sure to call `eth_requestAccounts` 
+      or `wallet_requestPermissions` first.
```

```suggestion
      with their account first, so make sure to call `eth_requestAccounts` first.
```

We recommend using `eth_requestAccounts` for all wallet restricted methods, so we don't need to specify `wallet_requestPermissions` in the description here. If the user clicks onto `eth_requestAccounts`, they should see that it calls `wallet_requestPermissions` internally.

---

I understand and I'll change the text. While we recommend using `eth_requestAccounts` as a best practice is it an absolute requirement for every scenario? Are instances where developers might choose to use `wallet_requestPermissions` for more granular control over permissions, without using `eth_requestAccounts`. I mean, just for continuity with the definition presented at the start of the file and to paint the whole picture shouldn't we explicitly mention `wallet_requestPermissions` here? The user will see it when they head to `eth_requestAccounts` and that might be slightly confusing.

---

If you want to include both, we should differentiate that `eth_requestAccounts` is the recommended method. It could be as simple as:

```md
...make sure to call `eth_requestAccounts` (recommended) or `wallet_requestPermissions` first.
```